### ISSUE TITLE
error message for displaying large image

### ISSUE THREAD

**Issue by eendebakpt** [vispy/vispy#758]
The error occurs with the vispy/examples/basics/scene/image.py example. If I increase the size of the image displayed I get the following error message:

RuntimeError: OpenGL got errors (Check before draw): GL_INVALID_VALUE

In the example below the code runs fine for nn=6000, but generate an error for nn=9000. Maybe the image size is limitation of my graphics card, but it would be helpful if the error message was a but more specific.

-------
import sys
from vispy import scene
from vispy import app
import numpy as np

canvas = scene.SceneCanvas(keys='interactive')
canvas.size = 800, 600
canvas.show()

view = canvas.central_widget.add_view()

nn=9000
img_data = np.random.normal(size=(2000, nn, 3), loc=128,  scale=50).astype(np.ubyte)
image = scene.visuals.Image(img_data, parent=view.scene)

view.camera.rect = (-10, -10, image.size[0]+20, image.size[1]+20)

if __name__ == '__main__' and sys.flags.interactive == 0:
    app.run()

**Comment by Eric89GXL** [comment]
Yeah, a better error message would be good. Thanks for the report. Feel free to open a PR if you have a good idea of how to fix it. We could try querying device capabilities at some point, but I'm not sure how robust it would be. At least when catching when the error occurred we could make clear what type of visual was creating the problem, which would help.

**Comment by rougier** [comment]
yes, and most probably the texure limite size is 8192 on our card.

**Comment by djhoese** [comment]
I'm able to reproduce this and yes it is the size of the texture. The difficulty in providing a good error message is that this check happens right before the draw. Since the GPU (OpenGL) is not giving us a good error message beyond "invalid value", there isn't much we can do. However, we could add another `gl.check_error` calls at the end of the method here: https://github.com/vispy/vispy/blob/c03e167b829bdefa0e4d10495cc0621d364c581a/vispy/gloo/glir.py#L1557

However, since this would happen for *every* upload to a texture I think it would have some not so great performance issues.

**Comment by kmuehlbauer** [comment]
@djhoese Can we check the GPU parameters beforehand? I mean at the time when a user tries to create a texture greater than the GPU can handle?

**Comment by djhoese** [comment]
Hm good idea. Here is how sys_info does it:

https://github.com/vispy/vispy/blob/94bd954f12d21f73a83f5a58a7e2ab9adc206cc8/vispy/util/config.py#L435-L436
